homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
+function ToDoList(){
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper');
+}
+ToDoList.prototype = {
+    start: function() {
+	this.fetchList(function(data) {
+	    this.updateList(data);
+	    this.drawList();
+	    this.preloadFirstNote();
+	}.bind(this));
+	window.addEventListener('click', function(event) {
+	    this.onNoteOpen(event);
+	}.bind(this));
+    },
 
-(function() {
+    onNoteOpen: function(event) {
+	if (event.target.classList.contains('note-title')) {
+	    var id = event.target.dataset.noteId;
+	    var content = this._listNoteContent[id];
+	    window.dispatchEvent(new CustomEvent('note-open',
+						 { detail: content }));
+	};
+    },
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+    preloadFirstNote: function() {
+	if (this._listNoteContent.length !== 0) {
```

Comment:
nit: It's better to put the condition outside a function wholly wrapped by only one branch.